lib/msf/core/modules/external/shim.rb

```diff
@@ -39,20 +41,35 @@ def self.common_check(meta = {})
   end
 
   def self.mod_meta_common(mod, meta = {}, ignore_options: [])
-    meta[:path]        = mod.path.dump
-    meta[:name]        = mod.meta['name'].dump
-    meta[:description] = mod.meta['description'].dump
-    meta[:authors]     = mod.meta['authors'].map(&:dump).join(",\n          ")
-    meta[:license]     = mod.meta['license'].nil? ? 'MSF_LICENSE' : mod.meta['license']
+    meta[:path]             = mod.path.dump
+    meta[:name]             = mod.meta['name'].dump
+    meta[:description]      = mod.meta['description'].dump
+    meta[:authors]          = mod.meta['authors'].map(&:dump).join(",\n          ")
+    meta[:license]          = mod.meta['license'].nil? ? 'MSF_LICENSE' : mod.meta['license']
+    meta[:options]          = mod_meta_common_options(mod, 'options', ignore_options: ignore_options)
+    meta[:advanced_options] = mod_meta_common_options(mod, 'advanced_options', ignore_options: ignore_options)
```

Comment:
This looks like you expect external modules to supply two separate lists of options. If we want to go the route of suppressing viewing of some options by default (I haven't been a fan of this paradigm) I would rather that be an attribute of the option and have a single list. An advanced option in Python might look like:
```python
'http_agent': {'type': 'string', 'description': '...', ..., 'advanced': True}
```

Reply:
Changed in commit 162139ee1d7c17b4de606f3ec007446353e66144. I've also updated the `pr_demo.py` example module.